api/openapi.yaml

```diff
@@ -1256,172 +1399,137 @@ components:
         A feature is a feature or service offered to a customer.
         For example: CPU-Hours, Tokens, API Calls, etc.
       allOf:
-        - $ref: "#/components/schemas/CreateFeatureRequest"
-        - type: object
-          required:
-            - id
-          properties:
-            id:
-              description: |
-                Readonly unique ULID identifier of the feature.
-              readOnly: true
-              type: string
-              example: 01ARZ3NDEKTSV4RRFFQ69G5FAV
-            createdAt:
-              description: |
-                The time the feature was created.
-              type: string
-              format: date-time
-              example: "2023-01-01T00:00:00Z"
-            updatedAt:
-              description: |
-                The time the feature was last updated.
-              type: string
-              format: date-time
-              example: "2023-01-01T00:00:00Z"
-    CreateLedger:
-      x-go-type-import:
-        path: github.com/openmeterio/openmeter/internal/credit
-      x-go-type: credit.Ledger
+        - $ref: "#/components/schemas/FeatureCreateInputs"
+        - $ref: "#/components/schemas/SharedMetaFields"
+    EntitlementCreateInputs:
       type: object
       description: |
-        A ledger represented in open meter. A ledger must be assigned to a single
-        subject.
+        Entitles a subject to use a feature.
       required:
-        - subject
+        - featureId
+      additionalProperties: false
       properties:
-        subject:
+        featureId:
```

Comment:
we should consider adding a `limitType: 'soft' | 'hard'` so later in SDKs we can decide behavior when balance=0